src/ensembl/src/content/app/custom-download/state/customDownloadSelectors.ts

```diff
@@ -0,0 +1,22 @@
+import { RootState } from 'src/store';
+
+export const getSelectedPreFilter = (state: RootState): string =>
+  state.customDownload.preFilter.selectedPreFilter;
+
+export const getShowPreFilterPanel = (state: RootState): boolean =>
+  state.customDownload.preFilter.showPreFiltersPanel;
+
+export const getSelectedTabButton = (state: RootState): string =>
+  state.customDownload.tabButton.selectedTabButton;
+
+export const getPreviewResult = (state: RootState): string =>
+  state.customDownload.resultHolder.preview;
+
+export const getIsLoadingResult = (state: RootState): boolean =>
+  state.customDownload.resultHolder.isLoadingResult;
```

Comment:
Regarding naming of things. Redux is a data layer; while such things as tabButton or resultHolder (if I understand correctly how they are used in the code) are interface elements. It's ok to keep ui-related information in the data layer, but isn't this information more appropriately described with words such as `results` or `tab`, without tight coupling to the interface elements that are used for displaying this data? Also, the `tab` property (a string?), instead of sitting in the `tabButton` part of the state (which is too concrete), can sit in the `ui` part of the state.

Reply:
That's done.